src/swap/defi/lifi.ts

```diff
@@ -0,0 +1,392 @@
+      toTokenId,
+      nativeAmount,
+      fromWallet,
+      toWallet,
+      quoteFor
+    } = request
+    if (quoteFor !== 'from') {
+      throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+    }
+
+    const fromToken = fromWallet.currencyConfig.allTokens[fromTokenId ?? '']
+    let fromContractAddress
+    let sendingToken = false
+    if (fromCurrencyCode === fromWallet.currencyInfo.currencyCode) {
+      fromContractAddress = fromCurrencyCode
+    } else {
+      sendingToken = true
+      fromContractAddress = fromToken?.networkLocation?.contractAddress
+    }
+
+    const toToken = toWallet.currencyConfig.allTokens[toTokenId ?? '']
+    let toContractAddress
+    if (toCurrencyCode === toWallet.currencyInfo.currencyCode) {
+      toContractAddress = toCurrencyCode
+    } else {
+      toContractAddress = toToken?.networkLocation?.contractAddress
+    }
+
+    if (fromContractAddress == null || toContractAddress == null) {
+      throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+    }
+
+    const { appId, integrator } = asInitOptions(opts.initOptions)
+    const fetch = io.fetch
+    // const fetch = nodeFetch
+
+    // Do not support transfer between same assets
+    if (
+      fromWallet.currencyInfo.pluginId === toWallet.currencyInfo.pluginId &&
+      request.fromCurrencyCode === request.toCurrencyCode
+    ) {
+      throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+    }
+
+    let lifiServers: string[] = LIFI_SERVERS_DEFAULT
+
+    checkInvalidCodes(INVALID_CURRENCY_CODES, request, swapInfo)
+
+    // Grab addresses:
+    const fromAddress = await getAddress(fromWallet)
+    const toAddress = await getAddress(toWallet)
+
+    const fromMainnetCode =
+      MAINNET_CODE_TRANSCRIPTION[fromWallet.currencyInfo.pluginId]
+    const toMainnetCode =
+      MAINNET_CODE_TRANSCRIPTION[toWallet.currencyInfo.pluginId]
+
+    if (fromMainnetCode == null || toMainnetCode == null) {
+      throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+    }
+
+    const now = Date.now()
+    if (
+      now - exchangeInfoLastUpdate > EXCHANGE_INFO_UPDATE_FREQ_MS ||
+      exchangeInfo == null
+    ) {
+      try {
+        const exchangeInfoResponse = await promiseWithTimeout(
+          fetchInfo(fetch, `v1/exchangeInfo/${appId}`)
+        )
+
+        if (exchangeInfoResponse.ok === true) {
+          exchangeInfo = asExchangeInfo(await exchangeInfoResponse.json())
+          exchangeInfoLastUpdate = now
+        } else {
+          // Error is ok. We just use defaults
+          log('Error getting info server exchangeInfo. Using defaults...')
+        }
+      } catch (e: any) {
+        log(
+          'Error getting info server exchangeInfo. Using defaults...',
+          e.message
+        )
+      }
+    }
+
+    if (exchangeInfo != null) {
+      const { lifi } = exchangeInfo.swap.plugins
+      lifiServers = lifi?.lifiServers ?? lifiServers
```

Comment:
Another nit pick: this block can be removed if you update the lifiServers array where you set 'exchangeInfoLastUpdate = now' in the try/catch